TrueDialog.Client.Standard/Helpers/Utils.cs

```diff
@@ -0,0 +1,50 @@
+﻿using System;
+using System.Text;
+
+namespace TrueDialog.Helpers
+{
+    public static class Utils
+    {
+        internal static string SoftReadPhoneNumber(string phoneNumber)
+        {
+            var rval = RemoveSpecialCharacters(phoneNumber);
+
+            if (rval[0] == '1' && rval.Length == 11)
+                return rval;
+            else if (rval[0] != '1' && rval.Length == 10)
+                rval = "1" + rval;
+            else
+                throw new ArgumentException("Wrong phone number format. Please follow E.164 format (+1xxxxxxxxxx).");
+
+            return rval;
+        }
+
+        internal static string ReadPhoneNumber(string phoneNumber)
+        {
+            var rval = RemoveSpecialCharacters(phoneNumber);
+
+            if (rval[0] == '1' && rval.Length == 11)
+                rval = "+" + rval;
+            else if (rval[0] != '1' && rval.Length == 10)
+                rval = "+1" + rval;
+            else
+                throw new ArgumentException("Wrong phone number format. Please follow E.164 format (+1xxxxxxxxxx).");
+
+            return rval;
+        }
+
+        internal static string RemoveSpecialCharacters(string str)
+        {
+            StringBuilder sb = new StringBuilder();
+            foreach (char c in str)
+            {
+                if (c >= '0' && c <= '9')
+                {
+                    sb.Append(c);
+                }
+            }
+
+            return sb.ToString();
```

Comment:
This looks like my initial implementation before I understood linq.  There's probably a cleaner way of doing this:
```csharp
return new String(str.Where(c => Char.IsDigit(c)).ToArray());  // Maybe?
```